### PR TITLE
TOML test suites

### PR DESCRIPTION
**Pull request by ben-albrecht**
This PR adds 2 test suites to the `test/modules/packages/Toml` directory:

- [specification](https://github.com/toml-lang/toml/tree/master/tests)
- [BurntSushi](https://github.com/BurntSushi/toml-test/tree/master/tests/valid)
  - `valid` tests only

Both test suites are futures for now, as the majority of tests fail.

Ideally, we'd have a future per execution option, but this is not yet supported in `start_test`

## Details

In supporting the BurntSushi test suite, I added a JSON writer to the TOML module.

The BurntSushi "good" files are provided as validation JSON files, and diffing is done comparing the JSON data structure produced by the TOML module. For now, this is handled with Python scripts (see `compare.py` and the caller, `PREDIFF`)

The specification tests do not provide validation JSON files, so I stuck to the provided TOML files as good files. We may want to generate the validation JSON files and use the same testing methodology for that suite eventually.

## Other changes

- Added `tomlType` method for getting the `tag` in a string format (compatible with the JSON writer)
- Added empty domain checks to fix case where parsing an empty toml file would produce an error
- Did some minor refactoring in `TOML.chpl`
   - Added `writeTOML()` which `writeThis()` calls
   - Made newlines more explicit in `writeTOML()`
- Hoisted the `re2` `SKIPIF` out to `Toml.skipif` for all `Toml/` tests

### Testing

- [x] `test/modules/packages/Toml`
    - `specification/` : `0/5`
    - `BurntSushi/` : `7/37`
- [x] `test/mason`

### CODE REVIEW

**Comment by ben-albrecht** [comment]
Side note: there were no big surprises in running these tests - most of the cases that these tests caught are already documented as TODO features in #7104

**Comment by ben-albrecht** [comment]
note about licensing: both projects use the `WTFPL` license

File: modules/packages/TOML.chpl [hunk 1]
```diff
@@ -76,6 +74,8 @@ module TomlParser {
 
   /* Prints a line by line output of parsing process */
   config const debugTomlParser = false;
+  /* Number of spaces in an indentation for JSON output */
+  config const tabSpace = 4;
```
Comment (benharsh) [inline modules/packages/TOML.chpl]: Maybe just make this a 'const' for now? 'tabSpace' seems like a config that could conflict with a lot of other packages.

Comment (ben-albrecht) [reply modules/packages/TOML.chpl]: Sure thing

**Pull request merged**